### ISSUE TITLE
`CantHandleQueryException` in `start n=node(*) ...` queries in 3.2

### ISSUE THREAD

**Issue by ayosec** [neo4j/neo4j#9452]
In Neo4j 3.2.0 Community, with default configuration, the old syntax `start n=node(*) return count(n)` throws a `CantHandleQueryException` error.

**Neo4j version**: 3.2.0
**Operating system**: Linux
**API/Driver**: Using HTTP interface

### Steps to reproduce

1. Launch a Neo4j instance

2. Execute a query with `start n=node(*) ...`

    ```bash
    $ curl $IP:7474/db/data/cypher                        \
        -H 'content-type: application/json'               \
        -d '{"query": "start n=node(*) return count(n)"}'
    ```

### Expected behavior

Execute the query and returns the count:

    {
      "columns" : [ "count(n)" ],
      "data" : [ [ ... ] ]
    }

### Actual behavior

Response from Neo4j when `cypher.default_language_version` option is `3.2`:

    {
      "message": "Internal error - should have used fall back to execute query, but something went horribly wrong",
      "exception": "CantHandleQueryException",
      "fullname": "org.neo4j.cypher.internal.ir.v3_2.exception.CantHandleQueryException",
      "stackTrace": [
        "org.neo4j.cypher.internal.compiler.v3_2.ast.convert.plannerQuery.ClauseConverters$$anonfun$addStartToLogicalPlanInput$1.apply(ClauseConverters.scala:471)",
        "org.neo4j.cypher.internal.compiler.v3_2.ast.convert.plannerQuery.ClauseConverters$$anonfun$addStartToLogicalPlanInput$1.apply(ClauseConverters.scala:460)",
        "org.neo4j.cypher.internal.ir.v3_2.PlannerQuery$class.amendQueryGraph(PlannerQuery.scala:66)",
        "org.neo4j.cypher.internal.ir.v3_2.RegularPlannerQuery.amendQueryGraph(PlannerQuery.scala:179)",
        "org.neo4j.cypher.internal.compiler.v3_2.ast.convert.plannerQuery.PlannerQueryBuilder$$anonfun$amendQueryGraph$1.apply(PlannerQueryBuilder.scala:34)",
        "org.neo4j.cypher.internal.compiler.v3_2.ast.convert.plannerQuery.PlannerQueryBuilder$$anonfun$amendQueryGraph$1.apply(PlannerQueryBuilder.scala:34)",
        "org.neo4j.cypher.internal.ir.v3_2.PlannerQuery$class.updateTailOrSelf(PlannerQuery.scala:81)",
        "org.neo4j.cypher.internal.ir.v3_2.RegularPlannerQuery.updateTailOrSelf(PlannerQuery.scala:179)",
        "org.neo4j.cypher.internal.compiler.v3_2.ast.convert.plannerQuery.PlannerQueryBuilder.amendQueryGraph(PlannerQueryBuilder.scala:34)",
    [...]

The same query works when option `cypher.default_language_version=3.1` is set in `neo4j.conf`

**Comment by systay** [comment]
This issue is solved by #9468

**Comment by ayosec** [comment]
@systay Thanks!
